core/src/rpc_subscriptions.rs

```diff
@@ -60,44 +61,49 @@ impl std::fmt::Debug for NotificationEntry {
     }
 }
 
+struct SubscriptionData<S> {
+    sink: Sink<S>,
+    commitment: CommitmentConfig,
+    last_notified_slot: RwLock<Slot>,
```

Comment:
Why do we need this to be `RwLock` wrapped?

Reply:
Hmm, maybe it doesn't. I thought we were using multiple threads for notifications, and was concerned about duplicate notifications. But looks like it's just one thread after all.

Reply:
Actually, I'm going to leave this so that we don't need to take a write lock on account subscriptions, and so it's ready for multi-threading if we need to use that in the future.